src/modeler/AutoRest.Swagger/Validation/ParameterDescriptionRequired.cs

```diff
@@ -1,34 +1,26 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using AutoRest.Core.Logging;
-using AutoRest.Core.Properties;
-using AutoRest.Swagger.Validation.Core;
 using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Validation.Core;
 
 namespace AutoRest.Swagger.Validation
 {
-    public class ParameterDescriptionRequired : TypedRule<SwaggerParameter>
+    public class ParameterDescriptionRequired : DescriptionRequired<SwaggerParameter>
     {
-        /// <summary>
-        /// This rule fails if the description is null and the reference is null (since the reference could have a description)
-        /// </summary>
-        /// <param name="entity"></param>
-        /// <returns></returns>
-        public override bool IsValid(SwaggerParameter entity)
-            => entity == null || entity.Description != null || !string.IsNullOrEmpty(entity.Reference);
+        private static readonly string ParameterTypeFormatter = "'{0}' parameter";
 
         /// <summary>
-        /// The template message for this Rule. 
-        /// </summary>
-        /// <remarks>
-        /// This may contain placeholders '{0}' for parameterized messages.
-        /// </remarks>
-        public override string MessageTemplate => Resources.MissingDescription;
-
-        /// <summary>
-        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// This rule fails if the description is null and the reference is null (since the reference could have a description)
         /// </summary>
-        public override Category Severity => Category.Warning;
+        /// <param name="entity">Entity being validated</param>
+        /// <param name="context">Rule context</param>
+        /// <param name="formatParameters">formatted parameters</param>
+        /// <returns><c>true</c> if entity contains description, <c>false</c> otherwise</returns>
+        public override bool IsValid(SwaggerParameter entity, RuleContext context, out object[] formatParameters)
+        {
+            formatParameters = new string[] { string.Format(ParameterTypeFormatter, entity.Name) };
+            return !string.IsNullOrWhiteSpace(entity?.Description) || !string.IsNullOrWhiteSpace(entity.Reference);
```

Comment:
@veronicagg I found that `parameters` in the operations can have refs. So i cannot remove this. 

ex: https://github.com/Azure/autorest/blob/master/src/modeler/AutoRest.Swagger.Tests/Resource/Swagger/Validation/positive/clean-complex-spec.json#L166